### ISSUE TITLE
[ToggleGroup] Deselecting requires two clicks if mapping to/from boolean values

### ISSUE THREAD

**Issue by andremonteiro95** [radix-ui/primitives#2254]
### Current Behavior

When mapping `ToggleGroup` values from, and to, boolean values, deselecting the value requires 2 clicks. It can also select the other value if you click them in this order: #1, #2,  and then #2 again.

This is for toggle groups with type "single".

### Expected behavior

Clicking the active value once should deselect it. 

### Reproducible example

[Stackblitz reproducible example](https://stackblitz.com/edit/stackblitz-starters-utqjfs?file=src%2FApp.tsx)

### Suggested solution

Not sure how to fix it. I tried investigating it and I found that the `value` returned by `useControllableState` in line 94 is not updated on the first click.

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) |    @radix-ui/react-toggle-group     |    1.0.4     |
| React            | n/a     |    18.2.0     |
| Browser          |     Chrome    |     114.0.5735.198    |
| Assistive tech   |         |         |
| Node             | n/a     |     16.17.0    |
| npm/yarn         |         |         |
| Operating System |         |         |

**Comment by andremonteiro95** [comment]
Exactly, I was able to work around it by setting it to `null`. However, the component doesn't allow for `null` as the type is `string | undefined`, it seems more intuitive to use `null` rather than an empty string.

**Comment by jamesratcliffe** [comment]
Yeah, it has to be `""` if you're using Typescript
